[2025-03-10T16:45:41.321+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: project2.insert_employees manual__2025-03-10T16:45:38.023190+00:00 [queued]>
[2025-03-10T16:45:41.335+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: project2.insert_employees manual__2025-03-10T16:45:38.023190+00:00 [queued]>
[2025-03-10T16:45:41.335+0000] {taskinstance.py:1279} INFO - 
--------------------------------------------------------------------------------
[2025-03-10T16:45:41.336+0000] {taskinstance.py:1280} INFO - Starting attempt 1 of 1
[2025-03-10T16:45:41.336+0000] {taskinstance.py:1281} INFO - 
--------------------------------------------------------------------------------
[2025-03-10T16:45:41.358+0000] {taskinstance.py:1300} INFO - Executing <Task(PostgresOperator): insert_employees> on 2025-03-10 16:45:38.023190+00:00
[2025-03-10T16:45:41.366+0000] {standard_task_runner.py:55} INFO - Started process 1054 to run task
[2025-03-10T16:45:41.371+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'project2', 'insert_employees', 'manual__2025-03-10T16:45:38.023190+00:00', '--job-id', '119', '--raw', '--subdir', 'DAGS_FOLDER/project2.py', '--cfg-path', '/tmp/tmpmwrbvsws']
[2025-03-10T16:45:41.371+0000] {standard_task_runner.py:83} INFO - Job 119: Subtask insert_employees
[2025-03-10T16:45:41.453+0000] {task_command.py:388} INFO - Running <TaskInstance: project2.insert_employees manual__2025-03-10T16:45:38.023190+00:00 [running]> on host 9f269e3d46f6
[2025-03-10T16:45:41.538+0000] {taskinstance.py:1509} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=project2
AIRFLOW_CTX_TASK_ID=insert_employees
AIRFLOW_CTX_EXECUTION_DATE=2025-03-10T16:45:38.023190+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2025-03-10T16:45:38.023190+00:00
[2025-03-10T16:45:41.539+0000] {sql.py:254} INFO - Executing: CALL update_employees();
[2025-03-10T16:45:41.554+0000] {base.py:73} INFO - Using connection ID 'postgres_conn' for task execution.
[2025-03-10T16:45:41.803+0000] {base.py:73} INFO - Using connection ID 'postgres_conn' for task execution.
[2025-03-10T16:45:41.834+0000] {sql.py:375} INFO - Running statement: CALL update_employees();, parameters: None
[2025-03-10T16:45:41.867+0000] {taskinstance.py:1768} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/operators/sql.py", line 266, in execute
    **extra_kwargs,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/hooks/sql.py", line 349, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/common/sql/hooks/sql.py", line 380, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.AmbiguousColumn: column reference "educationfield" is ambiguous
LINE 2: ...ECT s.employee_id, s.age, s.gender, s.OverTime, s,EducationF...
                                                             ^
QUERY:  INSERT INTO employees (employee_id, age, gender, OverTime, EducationField, PerformanceRating)
SELECT s.employee_id, s.age, s.gender, s.OverTime, s,EducationField, s.PerformanceRating
FROM staging_hr s
LEFT JOIN employees e ON s.employee_id = e.employee_id
WHERE e.employee_id IS NULL
CONTEXT:  PL/pgSQL function update_employees() line 3 at SQL statement

[2025-03-10T16:45:41.880+0000] {taskinstance.py:1323} INFO - Marking task as FAILED. dag_id=project2, task_id=insert_employees, execution_date=20250310T164538, start_date=20250310T164541, end_date=20250310T164541
[2025-03-10T16:45:41.907+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 119 for task insert_employees (column reference "educationfield" is ambiguous
LINE 2: ...ECT s.employee_id, s.age, s.gender, s.OverTime, s,EducationF...
                                                             ^
QUERY:  INSERT INTO employees (employee_id, age, gender, OverTime, EducationField, PerformanceRating)
SELECT s.employee_id, s.age, s.gender, s.OverTime, s,EducationField, s.PerformanceRating
FROM staging_hr s
LEFT JOIN employees e ON s.employee_id = e.employee_id
WHERE e.employee_id IS NULL
CONTEXT:  PL/pgSQL function update_employees() line 3 at SQL statement
; 1054)
[2025-03-10T16:45:41.945+0000] {local_task_job.py:208} INFO - Task exited with return code 1
[2025-03-10T16:45:42.013+0000] {taskinstance.py:2578} INFO - 0 downstream tasks scheduled from follow-on schedule check
